File: samples/java/bulk-sample/src/main/java/com/bhf/aeroncache/samples/BulkSample.java
package com.bhf.aeroncache.samples;

import com.bhf.aeroncache.client.AeronCacheClient;
import com.bhf.aeroncache.models.*;

import java.util.Arrays;
import java.util.UUID;

public class BulkSample {
    public static void main(String[] args) throws Exception {
        String baseUrl = args.length > 0 ? args[0] : "http://localhost:7070";
        String wsUrl = args.length > 1 ? args[1] : "ws://localhost:7071";

        System.out.println("Starting Java Bulk Operations Sample against " + baseUrl);

        AeronCacheClient client = new AeronCacheClient(baseUrl, wsUrl);
        String cacheId = "bulk-java-sample";

        BulkCacheOpsRequest bulkRequest = BulkCacheOpsRequest.builder()
                .requestId(UUID.randomUUID().toString())
                .operations(Arrays.asList(
                        CacheOperationRequest.builder()
                                .operationType(BulkOperationType.CREATE_CACHE)
                                .requestId("op-1")
                                .cacheId(cacheId)
                                .build(),
                        CacheOperationRequest.builder()
                                .operationType(BulkOperationType.ADD_ITEM)
                                .requestId("op-2")
                                .cacheId(cacheId)
                                .key("java-bulk-1")
                                .value("value-1")
                                .build(),
                        CacheOperationRequest.builder()
                                .operationType(BulkOperationType.GET_ITEM)
                                .requestId("op-3")
                                .cacheId(cacheId)
                                .key("java-bulk-1")
                                .build()
                ))
                .build();

        System.out.println("Executing bulk operations...");
        BulkCacheOpsResponse response = client.bulkOps(bulkRequest);

        System.out.println("Bulk Response ID: " + response.getRequestId());
        for (CacheOperationResponse opResp : response.getOperationResponses()) {
            System.out.printf("  Op %s: status=%s, cache=%s, key=%s, value=%s%n",
                    opResp.getRequestId(), opResp.getStatus(), opResp.getCacheId(), opResp.getKey(), opResp.getValue());
        }
    }
}

File: samples/python/bulk-sample/bulk_sample.py
import sys
import uuid
from aeron_cache.client import AeronCacheClient
from aeron_cache.models import BulkCacheOpsRequest, CacheOperationRequest, BulkOperationType


def main():
    base_url = "http://localhost:7070"
    ws_url = "http://localhost:7071"
    if len(sys.argv) > 1:
        base_url = sys.argv[1]
        ws_url = sys.argv[2]

    print(f"Starting Bulk Operations Sample against {base_url}")

    client = AeronCacheClient(base_url, ws_url)
    cache_id = "bulk-sample-cache"

    # Define a series of operations to be executed in bulk
    operations = [
        CacheOperationRequest(
            operationType=BulkOperationType.CREATE_CACHE,
            requestId="op-1",
            cacheId=cache_id
        ),
        CacheOperationRequest(
            operationType=BulkOperationType.ADD_ITEM,
            requestId="op-2",
            cacheId=cache_id,
            key="bulk-key-1",
            value="bulk-value-1"
        ),
        CacheOperationRequest(
            operationType=BulkOperationType.ADD_ITEM,
            requestId="op-3",
            cacheId=cache_id,
            key="bulk-key-2",
            value="bulk-value-2"
        ),
        CacheOperationRequest(
            operationType=BulkOperationType.GET_ITEM,
            requestId="op-4",
            cacheId=cache_id,
            key="bulk-key-1"
        )
    ]

    bulk_request = BulkCacheOpsRequest(
        requestId=str(uuid.uuid4()),
        operations=operations
    )

    print(f"Executing {len(operations)} operations in bulk...")
    response = client.bulk_ops(bulk_request)

    print(f"Bulk Request ID: {response.requestId}")
    for op_resp in response.operationResponses:
        print(f"  Operation {op_resp.requestId}: status={op_resp.status}, cache={op_resp.cacheId}, key={op_resp.key}, value={op_resp.value}")


if __name__ == "__main__":
    main()
